cognite/client/data_classes/data_modeling/ids.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+
+@dataclass
+class DataModelingId:
+    space: str
+    external_id: str
+
+    @classmethod
+    def from_tuple(cls, tup: tuple[str, str] | tuple[str, str, str]) -> DataModelingId:
+        return DataModelingId(*tup)
+
+
+@dataclass
+class VersionedDataModelingId(DataModelingId):
```

Comment:
The problem with this subclassing `DataModelingId` is that this indicates that e.g. `ContainerId` also can be a `VersionedDataModelingId`, which it can't.